### ISSUE TITLE
Consider making non-strict mode the default

### ISSUE THREAD

**Issue by jesskfullwood** [samscott89/serde_qs#21]
Hi. There is a discussion about your lib in https://github.com/actix/actix-web/issues/259. It seems that browsers (at least, Chromium and Firefox) ~~will not~~ EDIT *will* percent-encode an array from a select-multiple submission, which would seem to be an important use-case for this library. Therefore I suggest that `strict = false` should be the default, to be compatible with query strings from browsers without resorting to `Config`

**Comment by samscott89** [comment]
Hey! Thanks for looping me into the discussion. I'll try to chime in there later if I can be helpful.

It sounds like what you're describing is the current behaviour?
`cars[]=saab&cars[]=opel` -> `cars: vec!["saab", "opel]` is true in both strict and non-strict mode?
![image](https://user-images.githubusercontent.com/13355482/61642836-af113a80-ac6f-11e9-98e5-3d18a7bbd4c5.png)

In non-strict mode, you could additionally deserialize `cars%5B%5D=saab&cars%5B%5D=opel`. To the same thing. Are there browsers which would create that querystring for the example form?
(For comparison, strict mode would deserialize that as `"cars[]"="saab"`, `"cars[]"="opel"` and presumably end up overwriting the first entry.)

**Comment by jesskfullwood** [comment]
Ah, sorry I have it backwards. The request from the browser is `http://localhost:8080/select?cars%5B%5D=saab&cars%5B%5D=opel` i.e. it *is* percent-encoded. Actix does not attempt to percent-decode it when calling `req.query_sting()` hence the problem in the other issue.

**Comment by sonfangos** [comment]
Hi Sam! Thanks for jumping in so quickly. To flesh out what @jesskfullwood just wrote, here's an example of the issue that at least one person (me) is experiencing.

If this code is run:

```rust
use actix_web::{web, App, HttpRequest, HttpResponse, HttpServer};

fn index(_: HttpRequest) -> HttpResponse {
    HttpResponse::Ok()
        .body("<form action=\"select\" method=\"get\">
                <select name=\"cars[]\" multiple=\"multiple\">
                  <option value=\"volvo\">Volvo</option>
                  <option value=\"saab\">Saab</option>
                  <option value=\"opel\">Opel</option>
                  <option value=\"audi\">Audi</option>
                </select>
                <button type=\"submit\">Send</button>
              </form>")
}

fn select_cars(req: HttpRequest) -> String {
    req.query_string().to_string()
}

fn main() {
    HttpServer::new(|| App::new()
        .route("/", web::get().to(index))
        .route("/select", web::get().to(select_cars)))
        .bind("0.0.0.0:8080")                                                                                                                        
        .unwrap()                                                                                                                                    
        .run()                                                                                                                                       
        .unwrap();   
}
```

then what we see in the address bar is:

`http://0.0.0.0:8080/select?cars[]=saab&cars[]=audi`

while the querystring as displayed on the page (and seen by Actix) is:

`cars%5B%5D=saab&cars%5B%5D=audi`

which means that an error will be thrown unless strict is set to false in serde_qs. If we then simply refresh the page, serde_qs has no problem dealing with the (now percent decoded) querystring. At least, that has been my experience. When it comes to keys with square brackets, this means that Actix users will generally have to resort to Config and will, as I understand it, not be able to use the Actix feature, which is a shame, because that Actix feature is otherwise fab.

**Comment by samscott89** [comment]
Ah great, thanks for the info. Always good to hear realistic usage/behaviours.

It's kind of fascinating what the browser displays vs sends. I've reproed what you see with percent encodings. Even though the browser looks like it's not encoding, and following the same browser link again will send them unencoded.

That being said, I'm not overly keen on changing up the default now. I could see two possibilities: one is to set a "strict" feature flag as default which you can deactivate. The other is to allow config values through `QsQueryConfig`. Any preferences?

**Comment by sonfangos** [comment]
`QsQueryConfig` is where I went looking (failing to notice `Config` until Jess mentioned it to me), so that would be my vote, maybe with a little note on `serde_qs::actix::QsQuery` about why setting strict to false may sometimes be necessary in the context of Actix-Web.

Thanks again for looking into this so quickly. It's a fantastic crate.

**Comment by medwards** [comment]
We've run into this recently too.
Maybe this seems naive but can you store multiple types with `App::data` that are retrievable with `HttpRequest::app_data`?

If so then `QsQueryConfig` definitely seems like the place to put this. Then `QsQuery` would do more in `from_request` (namely grab some `Config` field out of `app_data` in addition to the error handler, and then use the `Config` to deserialize).

**Comment by samscott89** [comment]
This should be fixed now with #22. Thanks @fermeise for the PR, and everyone in the thread for the feedback :)

**Comment by cloud303-cholden** [comment]
For some reason the `axum` module isn't showing up in the docs. So if anyone comes here like me looking for how to change the strictness with Axum, there's an example here: [link](https://github.com/samscott89/serde_qs/blob/8a13b677f1b99ab78870f8731e75a7642f2d0530/src/axum.rs#L144).
